fpx/src/api/handlers/requestor.rs

```diff
@@ -0,0 +1,89 @@
+use std::{collections::BTreeMap, str::FromStr};
+
+use axum::{extract::State, Json};
+use http::{HeaderMap, HeaderName, HeaderValue, Method};
+
+use crate::{
+    api::errors::ApiServerError,
+    data::Store,
+    models::{self, RequestorError},
+};
+
+#[tracing::instrument(skip_all)]
+pub async fn execute_requestor(
+    State(store): State<Store>,
+    Json(payload): Json<models::RequestorRequestPayload>,
+) -> Result<Json<models::Response>, ApiServerError<RequestorError>> {
+    let tx = store.start_transaction().await?;
+
+    let request_body = payload.body.unwrap_or_default();
+    let request_headers = payload.headers.unwrap_or_default();
+
+    let request = handle_request(
+        &payload.method,
+        &payload.url,
+        &request_headers,
+        request_body.clone(),
+    );
+
+    let request_id = Store::request_create(
+        &tx,
+        payload.method.as_str(),
+        &payload.url,
+        &request_body,
+        request_headers.clone(),
+    )
+    .await?;
+
+    tx.commit().await?;
+
+    let response = request.await?;
+
+    let response_headers = prepare_headers(response.headers());
+    let response_status = response.status().as_u16();
+    let response_body = &response.text().await?;
+
+    Ok(Json(models::Response {
+        id: request_id,
+        status: response_status,
+        headers: response_headers,
+        body: Some(response_body.to_owned()),
+        url: payload.url,
+    }))
+}
+
+async fn handle_request(
+    request_method: &String,
+    url: &String,
+    headers: &BTreeMap<String, String>,
+    body: String,
+) -> Result<reqwest::Response, reqwest::Error> {
+    let request_method: Method = Method::from_bytes(request_method.as_bytes()).unwrap(); // TODO
+
+    let mut header_map = HeaderMap::new();
+
+    for (key, val) in headers.iter() {
+        if let Ok(header_name) = HeaderName::from_str(key.as_str()) {
+            if let Ok(header_value) = HeaderValue::from_bytes(val.as_bytes()) {
+                header_map.insert(header_name, header_value);
+            }
+        }
+    }
+
+    reqwest::Client::new()
```

Comment:
We can initialize a client in as a static and call that. That way we are able to reuse the same client and thus benefit from keep alives and the likes.

Reply:
And also make it lazy so it doesn't get initialized if it isn't being used: 

Something like this: https://github.com/fiberplane/monofiber/blob/3a5a25fb290e16725925ea59b4f90c343fbcbc57/api/src/webhooks/mod.rs#L36-L53